### PR TITLE
[api] Configuration#delayed_write_to_backend now uses ActiveJob

### PR DESCRIPTION
**Pull request by evanrolfe**
Part of this card: https://trello.com/c/Tx9d3bUb/467-5-p4-event-system-change-delayed-jobs-to-be-created-with-activejobs

### CODE REVIEW

File: src/api/app/models/configuration.rb [hunk 1]
```diff
@@ -96,7 +96,7 @@ def update_from_options_yml
   # the database or in migrations when there is no backend
   # running
   def delayed_write_to_backend
-    delay.write_to_backend
+    ConfigurationWriteToBackendJob.perform_later(id)
```
Comment (ChrisBr) [inline src/api/app/models/configuration.rb]: [Nitpick] Is the id necessary as Configuration should be a singleton anyway?

File: src/api/app/jobs/configuration_write_to_backend_job.rb [hunk 2]
```diff
@@ -0,0 +1,5 @@
+class ConfigurationWriteToBackendJob < ApplicationJob
+  def perform(configuration_id)
```
Comment (ChrisBr) [inline src/api/app/jobs/configuration_write_to_backend_job.rb]: No queue_as necessary?

File: src/api/app/models/configuration.rb (hunk 1 above)
Comment (evanrolfe) [inline src/api/app/models/configuration.rb]: `Configuration` is still based on a row in the database with an id, and `write_to_backend` is an instance method on the model so I need the id to call it. This implementation isn't any different than how it was before.

File: src/api/app/jobs/configuration_write_to_backend_job.rb (hunk 2 above)
Comment (evanrolfe) [inline src/api/app/jobs/configuration_write_to_backend_job.rb]: Nope, if `queue_as` isn't specified then ActiveJob will allocate it to the "default" queue.

**Comment by evanrolfe** [comment]
Thanks @ChrisBr

**Pull request merged**